pkg/ccl/backupccl/backupresolver/targets.go

```diff
@@ -458,59 +463,100 @@ func DescriptorsMatchingTargets(
 				alreadyExpandedDBs[prefix.Database.GetID()] = struct{}{}
 			}
 
+			// If the table pattern does not have explicit db AND explicit schema,
+			// try looking up the specified schema name in the current database.
+			// All tables in the specified schema should be backed up if the schema
+			// name can be resolved under the current DB, in particular, no tables
+			// outside of the specified schema should be backed up.
+			if !hasSchemaScope {
+				hasSchemaScope, _, err = r.LookupSchema(ctx, currentDatabase, scName)
```

Comment:
do we need this? We have already checked for a fully qualified all table select (ATS), so we now have two cases to differentiate.

`schema.*` and `db.*`. We can probably use `prefix` to differentiate these?

I think `schema.*` has prefix.CatalogName = false and prefix.SchemaName = true, whereas `db.*` has both set to true. So we can do something like:

if !hasSchemaScope {
  // schema.* comment
  if !prefix.CatalogName && prefix.SchemaName {
        hasSchemaScope = true
   }
 // db.* comment
}